openedx/adg/lms/student/views.py

```diff
@@ -64,3 +73,120 @@ def get_social_oauth_providers_data(self):
                     })
 
         return context
+
+
+class DashboardView(LoginRequiredMixin, TemplateView):
+    """
+    View for dashboard page
+    """
+
+    template_name = 'student_account/dashboard.html'
+
+    def get_context_data(self, **kwargs):
+        """
+        Add the context regarding social auth providers to the view
+        """
+        context = super(DashboardView, self).get_context_data(**kwargs)
+
+        completed_courses_details, inprogress_courses_details = self.get_courses_overview_and_status()
+
+        webinars = self.get_webinars_overview()
+
+        certificates = self.get_certificates_overview()
+
+        context.update({
+            'user': self.request.user,
+            'completed_courses': completed_courses_details,
+            'inprogress_courses': inprogress_courses_details,
+            'webinars': webinars,
+            'certificates': certificates,
+        })
+        return context
+
+    def get_courses_overview_and_status(self):
```

Comment:
@faizan-philu - Can we move these three methods outside from views. 
In views we shouldn't write any business logic to get data. Please move this to helper files or any other appropriate place